src/Sarif.Sarifer/IBackgroundAnalysisService.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    /// <summary>
+    /// Interface for performing multiple static analyses in the background.
+    /// </summary>
+    internal interface IBackgroundAnalysisService
+    {
+        /// <summary>
+        /// Begins background analysis of the specified text.
+        /// </summary>
+        /// <param name="path">
+        /// The absolute path of the file being analyzed, or null if the text came from a VS text
+        /// buffer that was not attached to a file.
+        /// </param>
+        /// <param name="text">
+        /// The text to analyze.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Task"/> that represents the completion of the background analysis.
+        /// </returns>
+        Task StartAnalysisAsync(string path, string text);
```

Comment:
Should probably take a CancellationToken

Reply:
Same comment applies to all Async methods unless it really cannot be cancelled.

---
In reply to: [524491510](https://github.com/microsoft/sarif-visualstudio-extension/pull/266#discussion_r524491510) [](ancestors = 524491510)

Reply:
Cancellation is now plumbed from text view creation listener all the way down into the derived analyzer classes. Will also hook it up to the task center.

---
In reply to: [524491811](https://github.com/microsoft/sarif-visualstudio-extension/pull/266#discussion_r524491811) [](ancestors = 524491811,524491510)